src/update_engine/libcurl_http_fetcher.h

```diff
@@ -101,6 +101,13 @@ class LibcurlHttpFetcher : public HttpFetcher {
     check_certificate_ = check_certificate;
   }
 
+  // Sets HTTP Auth user and password for HTTPS connections.
```

Comment:
where is https forced, when auth_user_ is set?

Reply:
Auth is only enabled for HTTPS (in this function here https://github.com/flatcar-linux/update_engine/pull/7/files#diff-97693d3d897607cd79e4d24401d6515aL184) but ignored for HTTP

Reply:
oh oh, i read that the other way around.
So, if an unaware user was attempting to _test_ auth, against an update service (say in testing/dev) that is http://, then it will silently ignore it?
Perhaps there be a WARN output if http and `(!auth_user_.empty())`?

Reply:
The update service itself can be a development HTTP endpoint which gives out the HTTPS link to the payload. The password is only used for the payload download and only if it is an HTTPS link.

Reply:
Yes, a warning is good.

Reply:
Updated the PR